src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -0,0 +1,156 @@
+                                  Include="%(_ClonedProjectReference.Identity)" />
+        <!-- Populate items first with the project references so that we can trim them but keep the item metadata -->
+        <_PackageProjectReference Include="@(_ClonedProjectReference)" />
+        <_ClonedProjectReference Remove="@(_RemovePackageProjectReference)" />
+        <_PackageProjectReference Remove="@(_ClonedProjectReference)" />                                
+    </ItemGroup>
+  </Target>
+
+  <!-- Given a package drop, or list of package drops, retrieve a list of assembly names and their associated versions -->
+  <Target Name="GatherAssemblyVersionsFromPackagesDrops">
+    <GatherAssemblyVersionsFromPackageDrop PackageDrops="@(_PackagesDrops)">
+       <Output TaskParameter="AssemblyNames" ItemName="_AssemblyVersionList" />                                                  
+    </GatherAssemblyVersionsFromPackageDrop>                                           
+  </Target>
+
+  <Target Name="GatherProjectReferenceForProjectJson"
+          Condition="'$(KeepAllProjectReferences)' != 'true'"
+          Inputs="%(_PackageProjectReference.Identity)"
+          Outputs="fake">
+    <PropertyGroup>
+      <_PackageReferenceName>%(_PackageProjectReference.PackageReferenceName)</_PackageReferenceName>
+      <_InjectProjectReference>%(_PackageProjectReference.FileName)</_InjectProjectReference>
+
+      <!-- determine the version of the assembly in the project reference by looking at the packages in "PackagesDrops".  -->
+      <!-- If an alternate packagereferencename was specified, then look for that version number instead of the version number for the project reference. -->
+      <_AssemblyVersion Condition="'$(_PackageReferenceName)' != '' and '%(_AssemblyVersionList.AssemblyName)' == '$(_PackageReferenceName)'">%(_AssemblyVersionList.AssemblyVersion)</_AssemblyVersion>
+      <_AssemblyVersion Condition="'$(_AssemblyVersion)' == '' and '%(_AssemblyVersionList.AssemblyName)' == '$(_InjectProjectReference)'">%(_AssemblyVersionList.AssemblyVersion)</_AssemblyVersion>
+    </PropertyGroup>
+
+    <Message Condition="'$(_AssemblyVersion)' != ''" Text="Derived version number from packages drop." />   
+
+    <ItemGroup>
+      <_AdditionalProperties Include="%(_PackageProjectReference.Identity)">
+        <AssemblyVersionTransition>$(AssemblyVersionTransition)</AssemblyVersionTransition>
+      </_AdditionalProperties>
+    </ItemGroup>
+    <!-- Evaluate the project references to determine the assembly version for the package dependency.
+         Only evaluate the project reference if we were not able to find an assembly version in "PackageDrops" -->
+    <EvaluateMSBuildProject Condition="'%(_PackageProjectReference.Identity)' != '' and '$(_AssemblyVersion)' == ''"
```

Comment:
Is there any reason we cannot just call MSBuild task directly on this project and get the AssemblyVersions? I believe @ericstj is doing that in a number of places in packaging targets already.